fs.go

```diff
@@ -42,6 +42,13 @@ type Filesystem interface {
 	Base() string
 }
 
+// Symlinker is a Filesystem with support for creating symlinks.
+type Symlinker interface {
+	Filesystem
+	Symlink(oldname, newname string) error
```

Comment:
Document this methods.

Later, when implementing the interface, if the implementation is simple, just reference this documentation by writing something along the lines of `Symlink implements billy.Symlinker`.